5_analyze_data/notebooks/Heatmap_analysis/nf1_complexheatmap.r

```diff
@@ -0,0 +1,155 @@
+suppressPackageStartupMessages(library(ComplexHeatmap))
+suppressPackageStartupMessages(library(dplyr))
+
+# Set paths and constants
+input_data_dir <- file.path("..", "..", "..", "4_processing_features", "data")
+output_figure_dir <- "figures"
+
+cp_heatmap_file <- file.path(output_figure_dir, "cp_complex_heatmap.pdf")
+dp_heatmap_file <- file.path(output_figure_dir, "dp_complex_heatmap.pdf")
+
+# Set heatmap colors
+well_cols = c(
+    "C6" = "#E1DAAE",
+    "C7" = "#FF934F",
+    "D6" = "#CC2D35",
+    "D7" = "#058ED9",
+    "E6" = "#848FA2",
+    "E7" = "#2D3142",
+    "F6" = "#FFC857",
+    "F7" = "#5f7a12"
+)
+genotype_cols = c(
+    "Null" = "#785EF0",
+    "WT" = "#DC267F"
+)
+
+# Load data
+cp_file <- file.path(input_data_dir, "nf1_sc_norm_fs_cellprofiler.csv.gz")
+
+cp_df <- readr::read_csv(
+    cp_file,
+    col_types = readr::cols(
+        .default="d",
+        Metadata_WellRow="c",
+        Metadata_WellCol="c",
+        Metadata_Well="c",
+        Metadata_gene_name="c",
+        Metadata_genotype="c"
+    )
+) %>% dplyr::select(-...1)  # Drop index col
+
+print(dim(cp_df))
+head(cp_df, 3)
+
+# Split metadata and feature data
+cp_metadata_df <- cp_df %>% dplyr::select(tidyr::starts_with("Metadata"))
+cp_meta_cols <- colnames(cp_metadata_df)
+cp_df <- cp_df %>% dplyr::select(-!!cp_meta_cols)
+
+# Calculate correlation matrix from feature data
+cp_cor_matrix <- t(cp_df) %>% cor()
+
+print(dim(cp_cor_matrix))
+head(cp_cor_matrix, 3)
+
+ht <- Heatmap(
```

Comment:
So since you make a correlation matrix in R and then make a heatmap, why is the one that I created in Python so different to this one? I see there is a lot of code to make this heatmap so is R better equipped to make a better correlation heatmap?

Reply:
The scale in python is different from the scale in R. ComplexHeatmap auto scales, which is a caveat to interpretation.

It's also possible that the dendrogram ordering is different in python